packages/styled-base/types/index.d.ts

```diff
@@ -24,127 +23,88 @@ export {
   ObjectInterpolation
 } from '@emotion/serialize'
 
-export { ComponentSelector, Interpolation }
-
-type JSXInEl = JSX.IntrinsicElements
-
-export type WithTheme<P, T> = P extends { theme: infer Theme }
-  ? P & { theme: Exclude<Theme, undefined> }
-  : P & { theme: T }
+export { ComponentSelector, Interpolation, PropsOf, Omit }
 
 export interface StyledOptions {
   label?: string
   shouldForwardProp?(propName: string): boolean
   target?: string
 }
 
-export interface StyledComponent<InnerProps, StyleProps, Theme extends object>
-  extends React.SFC<InnerProps & StyleProps & { theme?: Theme }>,
+/**
+ * @typeparam ComponentProps  Props which will be included when withComponent is called
+ * @typeparam SpecificComponentProps  Props which will *not* be included when withComponent is called
+ */
+export interface StyledComponent<ComponentProps, SpecificComponentProps>
```

Comment:
this requires 2 type arguments now, isn't `ComponentProps` enough for majority of use cases? especially that `SpecificComponentProps` are somewhat purely to support `withComponent` which in general is a discouraged API?

In other words - WDYT about adding a default `{}` to  `SpecificComponentProps`?

Reply:
Default makes sense, will add

Reply:
Done